src/Microsoft.Diagnostics.Monitoring.RestServer/DiagnosticServices.cs

```diff
@@ -114,82 +112,46 @@ private static DumpType MapDumpType(Models.DumpType dumpType)
             }
         }
 
-        public async Task<IProcessInfo> GetProcessAsync(ProcessKey? processKey, CancellationToken token)
+        public async Task<IEnumerable<IProcessInfo>> GetProcessesAsync(ProcessFilterOptions processFilterConfig, CancellationToken token)
         {
-            var processInfos = await GetProcessesAsync(token);
+            IEnumerable<IProcessInfo> processes = await GetProcessesAsync(token);
 
-            if (processKey.HasValue)
+            if (processFilterConfig != null)
             {
-                return GetSingleProcessInfo(
-                    processInfos,
-                    processKey);
+                processes = processes.Where(p => processFilterConfig.ProcessFilters.All(c => ProcessFilterOptionsUtils.MatchFilter(c, p)));
```

Comment:
I think this should be folded into `Task<IEnumerable<IProcessInfo>> GetProcessesAsync(CancellationToken token)` as a extra filtering step and then add the ProcessFilterOptions processFilterConfig parameter to the method. I don't think anything is calling that method other than this one and the separation doesn't seem necessary.